src/modules/categories/CategoriesList.tsx

```diff
@@ -0,0 +1,36 @@
+'use client';
+
+import { type FC } from 'react';
+
+import { useQuery } from '@tanstack/react-query';
+
+import Error from '@/ui/error/Error';
+
+import './styles/list.scss';
+
+import { fetchCategories } from './api/fetchCategories';
+import { CategoryItem } from './types/CategoryItem';
+import CategoriesItem from './ui/CategoriesItem';
+
+const CategoriesList: FC = () => {
+  const { data, error } = useQuery<CategoryItem[]>({
+    queryKey: ['categories'],
+    queryFn: fetchCategories,
+  });
+
+  if (error || !data) {
+    const message =
+      error instanceof Error ? error.message : 'Something went wrong';
+    return <Error error={message} />;
+  }
+
+  return (
+    <ul className="list">
+      {data?.map((category, i) => (
```

Comment:
```suggestion
      {data.map((category, i) => (
```

Если ты обрабатываешь уже сверху, что у тебя нет даты, ts сам видит, что дальше дата 100% будет, и можно не ставить `?` везде